extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ClassConfigPropertiesUtil.java

```diff
@@ -0,0 +1,359 @@
+        } else {
+            createSetterCall(bytecodeCreator, configObject, setter, value);
+        }
+    }
+
+    private static void createSetterCall(BytecodeCreator bytecodeCreator, ResultHandle configObject,
+            MethodInfo setter, ResultHandle value) {
+        bytecodeCreator.invokeVirtualMethod(
+                MethodDescriptor.of(setter),
+                configObject, value);
+    }
+
+    private static void createFieldWrite(BytecodeCreator bytecodeCreator, ResultHandle configObject,
+            FieldInfo field, ResultHandle value) {
+        bytecodeCreator.writeInstanceField(FieldDescriptor.of(field), configObject, value);
+    }
+
+    private static boolean shouldCheckForDefaultValue(ClassInfo configPropertiesClassInfo, FieldInfo field) {
+        if (field.type().kind() == Type.Kind.PRIMITIVE) {
+            return false;
+        }
+        String getterName = JavaBeanUtil.getGetterName(field.name(), field.type().name().toString());
+        MethodInfo getterMethod = configPropertiesClassInfo.method(getterName);
+        if (getterMethod != null) {
+            return Modifier.isPublic(getterMethod.flags());
+        }
+
+        return !Modifier.isFinal(field.flags()) && Modifier.isPublic(field.flags());
+    }
+
+    /**
+     * Create code that uses the validator in order to validate the entire object
+     * If errors are found an IllegalArgumentException is thrown and the message
+     * is constructed by calling the previously generated VIOLATION_SET_TO_STRING_METHOD
+     */
+    private static void createValidationCodePath(MethodCreator bytecodeCreator, ResultHandle configObject,
+            String configPrefix) {
+        ResultHandle validationResult = bytecodeCreator.invokeInterfaceMethod(
+                MethodDescriptor.ofMethod(VALIDATOR_CLASS, "validate", Set.class, Object.class, Class[].class),
+                bytecodeCreator.getMethodParam(1), configObject,
+                bytecodeCreator.newArray(Class.class, bytecodeCreator.load(0)));
+        ResultHandle constraintSetIsEmpty = bytecodeCreator.invokeInterfaceMethod(
+                MethodDescriptor.ofMethod(Set.class, "isEmpty", boolean.class), validationResult);
+        BranchResult constraintSetIsEmptyBranch = bytecodeCreator.ifNonZero(constraintSetIsEmpty);
+        constraintSetIsEmptyBranch.trueBranch().returnValue(configObject);
+
+        BytecodeCreator constraintSetIsEmptyFalse = constraintSetIsEmptyBranch.falseBranch();
+
+        ResultHandle exception = constraintSetIsEmptyFalse.newInstance(
+                MethodDescriptor.ofConstructor(CONSTRAINT_VIOLATION_EXCEPTION_CLASS, Set.class.getName()), validationResult);
+        constraintSetIsEmptyFalse.throwException(exception);
+
+        constraintSetIsEmptyBranch.falseBranch().throwException(IllegalArgumentException.class,
+                "Configuration group " + configPrefix + " contains invalid values. The errors are: ");
```

Comment:
Hmmm, this should go away, shouldn't it?

Reply:
Awesome catch!